### PR TITLE
Hide inaccurate annotation counts when in user-focused mode

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR makes one small user-facing change (for an extremely small number of users, anyway!). As our methods of filtering annotations get more complex—we recently added a user-focused mode, e.g.—so does managing the representation and management of various filters in the client UI.

The immediate problem to fix—which this PR does remedy—is that the combination of a user-focused mode **and** "selected annotations" (e.g. clicking on annotated/highlighted text in the host page/document) can result in an inaccurate count of the annotations that _would_ be visible if the selection were cleared.

This PR also does some refactoring of the `SearchStatusBar` component to try to make its tri-mode nature a little more clear and code-document how one gets into different states/contexts.

To illustrate...

In a non-user-focused mode (i.e. the vast majority of our users' context), clicking on annotated text in the host page will drill the sidebar down to just annotations that target that text, e.g.:

annotation-counts-selection-mode

Note that the button to clear the "selection" filtering displays the number of annotations that will be displayed if the button is clicked (5 in this case). The number displayed there is a simple number: the total number of annotations in the root thread.

However, there are problems when "selection" filtering gets combined with user-mode filtering—that is, when we have multiple filters being applied on the displayed annotations. To wit:

annotation-counts-inaccurate

Note how the clear-selection button here claims that 14 annotations will be displayed if clicked, but, in fact, only 9 are. That's because it fails to take into account the remaining user filter.

Solution in this PR: hide the all-annotations count when also in focused-user mode, e.g.:

accurate-annotation-counts

### Edited

Thanks to a nice suggestion from @LMS007 , the clear-selection button text when in user-focused mode has been amended to show the focused user's name:

image

The rest of the interaction is the same as the video above.

The UX in this area is complex. I know it's on the strategic roadmap to work on the search and filtering UI in general.

Justification for solution here is:

1. Deriving the correct number to display when selection and user-focus modes are combined is non-trivial, and;
1. This state is quite rare—you'd have to be an instructor in grading mode and know that you can click on annotated text to further filter, and;
1. It feels like getting this _just_ right involves a larger design project

Thus: this PR makes it so we're not displaying numbers that are actually wrong. Hiding the annotation count when in user-focused mode is the entirety of the user-visible change here.

### CODE REVIEW

File: src/sidebar/components/search-status-bar.js [hunk 1]
```diff
@@ -24,113 +22,161 @@ const countVisibleAnns = annThread => {
...
+        case uiConstants.TAB_NOTES:
+          selectedText = 'Show all notes';
+          if (counts.notes > 1 && !modes.focused) {
+            selectedText += ` (${counts.notes})`;
+          }
+          break;
+        case uiConstants.TAB_ANNOTATIONS:
+        default:
+          selectedText = 'Show all annotations';
+          if (counts.annotations > 1 && !modes.focused) {
```
Comment (LMS007) [inline src/sidebar/components/search-status-bar.js]: What do you think about changing this a bit to include the focused user's name on this button when its in focused mode?

e.g. 

_Show all [blank's] annotations_

File: src/sidebar/components/search-status-bar.js [hunk 2]
```diff
@@ -24,113 +22,161 @@ const countVisibleAnns = annThread => {
...
-  const areNotAllAnnotationsVisible = () => {
-    if (storeState.directLinkedGroupFetchFailed) {
-      return true;
-    }
-    const selection = storeState.selectedAnnotationMap;
-    if (!selection) {
-      return false;
-    }
-    return Object.keys(selection).length > 0;
+  const btnProps = {
```
Comment (LMS007) [inline src/sidebar/components/search-status-bar.js]: nice consolidation

File: src/sidebar/components/search-status-bar.js [hunk 3]
```diff
@@ -24,113 +22,161 @@ const countVisibleAnns = annThread => {
...
+          selectedText = 'Show all annotations and notes';
+          break;
+        case uiConstants.TAB_NOTES:
+          selectedText = 'Show all notes';
+          if (counts.notes > 1 && !modes.focused) {
+            selectedText += ` (${counts.notes})`;
+          }
+          break;
+        case uiConstants.TAB_ANNOTATIONS:
+        default:
```
Comment (LMS007) [inline src/sidebar/components/search-status-bar.js]: nit:

my preference is not including a `default` if its invalid condition. e.g. can `selectedTab` ever not be one `TAB_ORPHANS`,  `TAB_NOTES` or `TAB_ANNOTATIONS` ?

**Comment by lyzadanger** [comment]
> Do you think long term is might be good to create a helper function inside build-thread that would expose to actual render annotation count (post filters)

And how. Yes.

File: src/sidebar/components/search-status-bar.js (hunk 3 above)
Comment (lyzadanger) [reply src/sidebar/components/search-status-bar.js]: Technically, there's nothing to _stop_ `selectedTab` from being any value (I don't think?). I was trying to communicate that the `annotations` tab is the "most likely" active tab, but I can kill the `default`.

File: src/sidebar/components/search-status-bar.js (hunk 1 above)
Comment (lyzadanger) [reply src/sidebar/components/search-status-bar.js]: I highly like that idea. Helps with the confusion in the UX.

**Pull request merged**